spec/unit/lcm/lcm2_spec.rb

```diff
@@ -45,10 +37,111 @@
       allow(domain).to receive(:data_products)
     end
 
-    it 'skips actions in skip_actions' do
-      expect(GoodData::LCM2::CollectSegments).not_to receive(:call)
-      expect(GoodData::LCM2::SynchronizeUsers).not_to receive(:call)
-      GoodData::LCM2.perform('users', params)
+    context 'when skip_actions specified' do
+      let(:params) do
+        params = {
+          skip_actions: %w(CollectSegments SynchronizeUsers),
+          GDC_GD_CLIENT: client,
+          GDC_LOGGER: logger,
+          domain: domain
+        }
+        GoodData::LCM2.convert_to_smart_hash(params)
+      end
+
+      it 'skips actions in skip_actions' do
+        expect(GoodData::LCM2::CollectSegments).not_to receive(:call)
+        expect(GoodData::LCM2::SynchronizeUsers).not_to receive(:call)
+        GoodData::LCM2.perform('users', params)
+      end
+    end
+
+    context 'when mandatory params are given and hello action is performed' do
+      let(:params) do
+        params = {
+          GDC_LOGGER: logger,
+          message: 'Ahoj'
+        }
+        GoodData::LCM2.convert_to_smart_hash(params)
+      end
+
+      it 'should finish successfully' do
+        result = GoodData::LCM2.perform('hello', params)
+        pp result
+        expect(result[:actions]).to eq(['HelloWorld'])
+        expect(result[:results]["HelloWorld"][0][:message]).to eq('Ahoj')
+        expect(result[:params][:message]).to eq('Ahoj')
+        expect(result[:params][:gdc_logger]).to eq(logger)
+        expect(result[:params][:iterations]).to eq(1)
+      end
+    end
+
+    context 'when invalid mode is specified' do
+      let(:params) do
+        params = {
+          GDC_LOGGER: logger
+        }
+        GoodData::LCM2.convert_to_smart_hash(params)
+      end
+
+      it 'should throw exception' do
+        expect do
```

Comment:
warning: Similar code found in lcm2_spec.rb:87, lcm2_spec.rb:132 (mass = 34) (flay)